src/types.ts

```diff
@@ -599,6 +600,28 @@ export class ZodString extends ZodType<string, ZodStringDef> {
           });
           status.dirty();
         }
+      } else if (check.kind === "utc") {
+        const parsedDate = new Date(input.data);
+
+        const isDate = !isNaN(parsedDate.getTime());
+        const isMsString = isDate && parsedDate.toISOString() === input.data;
```

Comment:
Interesting approach but I think this is better implemented with regexes for the "ISO with ms" and "ISO without ms".